bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -724,69 +730,63 @@ public Image(Device device, ImageFileNameProvider imageFileNameProvider) {
 public Image(Device device, ImageDataProvider imageDataProvider) {
 	super(device);
 	this.imageDataProvider = imageDataProvider;
-	currentNativeZoom = DPIUtil.getNativeDeviceZoom();
+	initialNativeZoom = DPIUtil.getNativeDeviceZoom();
 	ElementAtZoom<ImageData> data =  DPIUtil.validateAndGetImageDataAtZoom(imageDataProvider, getZoom());
-	ImageData resizedData = DPIUtil.autoScaleImageData(device, data.element(), data.zoom());
-	init (resizedData);
+	ImageData resizedData = DPIUtil.autoScaleImageData(device, data.element(), getZoom(), data.zoom());
+	init (resizedData, getZoom());
 	init();
 }
 
 /**
  * Update zoom and refresh the Image based on the native zoom level, if required.
  *
- * @param nativeZoom native zoom in % of the monitor on which the image is painted
+ * @param image the image to get the handle of
+ * @param deviceZoomLevel device zoom in % of the monitor on which the image is painted
  *
  * @return true if image is refreshed
+ * @since 3.126
  */
-boolean handleDPIChange (int nativeZoom) {
-	int newZoom = DPIUtil.getZoomForAutoscaleProperty(nativeZoom);
-	int oldZoom = this.getZoom();
-	boolean refreshed = false;
-	if (imageFileNameProvider != null) {
-		if (newZoom != oldZoom) {
-			ElementAtZoom<String> filename = DPIUtil.validateAndGetImagePathAtZoom (imageFileNameProvider, newZoom);
-			if (filename.zoom() == newZoom) {
-				/* Release current native resources */
-				destroy ();
-				initNative(filename.element());
-				if (this.handle == 0) init(new ImageData (filename.element()));
-				init();
-				refreshed = true;
-			} else {
-				/* Release current native resources */
-				destroy ();
-				ImageData resizedData = DPIUtil.autoScaleImageData (device, new ImageData (filename.element()), filename.zoom());
-				init(resizedData);
-				init ();
-				refreshed = true;
-			}
-			setCurrentNativeZoom(nativeZoom);
-		}
-	} else if (imageDataProvider != null) {
-		if (newZoom != oldZoom) {
-			ElementAtZoom<ImageData> data = DPIUtil.validateAndGetImageDataAtZoom (imageDataProvider, newZoom);
+public static Long win32_getHandle (Image image, int deviceZoomLevel) {
```

Comment:
Lets rename deviceZoomLevel to zoom for consistency

Reply:
Done